### DISCUSSION TITLE
Issue with deploying Backend Express API to Vercel

### DISCUSSION THREAD

**Discussion by Harriet27** [vercel/community#508] [Help] [answered]
Hi, I just deployed my Express API app to Vercel, and I encountered this error.
As it says, I require to install the `pg` package manually, although I've installed `pg` locally on the root project and globally.

Here's my `vercel.json`
```
{
  "rewrites": [
    {
      "source": "/api/(.*)",
      "destination": "/api"
    }
  ]
}
```
and `serverless.yml`
```
custom:
  webpack:
    includeModules:
      forceInclude:
        - pg
```
Also, when I visit the api url, it gives an error code 500, with message serverless function has crashed.
![image](https://user-images.githubusercontent.com/48342510/159126522-b826b7b8-b25e-459a-bfcd-99e9bb0804b8.png)
![image](https://user-images.githubusercontent.com/48342510/159126547-7bf55825-187c-45aa-9b37-d7bea6ab2e6b.png)

**Comment by leerob** [comment, marked as answer]
Hey! I would recommend using API Routes instead. Check out this article for more info. 

https://vercel.com/guides/using-express-with-vercel#using-express.js-with-vercel

**Reply by Harriet27** [reply]
I've tried using the API Route sand it gives me an error code 500, but if I didn't use the API Routes, it gives an error code of 404.
Which I supposed using API Routes already 1 step ahead.
![image](https://user-images.githubusercontent.com/48342510/159126972-05e95263-7bdb-4c36-b626-e42590fd1d45.png)

**Reply by leerob** [reply]
Are you only hosting a backend? Or a frontend as well? 

Here's an example using Prisma, which means you could easily switch between MySQL and Postgres if you wanted to: https://github.com/vercel/next.js/tree/canary/examples/with-mysql

And here's an example using Postgres with Supabase: https://github.com/vercel/next.js/tree/canary/examples/with-supabase-auth-realtime-db

**Reply by Harriet27** [reply]
Only backend, using Express with database of Postgresql.
I've successfully deployed my frontend NextJS on to Vercel, I'm just having issue with deploying my backend.

**Reply by leerob** [reply]
Check out this guide! https://vercel.com/guides/nextjs-prisma-postgres

**Reply by Harriet27** [reply]
O wait, does this works with gitlab? I'm using gitlab btw, sorry, should've tell in the first place 😅

**Reply by leerob** [reply]
Yeah, Vercel supports GitHub, GitLab, BitBucket, and others using the CLI / API 😄

**Reply by Harriet27** [reply]
Ah, lemme give the guide a shot!
